sql: add lazy check for transaction (#963)
pingcap/docs: sql/transaction.md

```sql
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED;
```
## Lazy check of transactions

> In TiDB, lazy checking is performed for values written by ordinary `INSERT` statements. The meaning of a lazy check is that: it does not check for unique constraints when the `INSERT` statement is executed, but checks for unique constraints when the transaction commits.

Reviewer: ```suggestion
**Lazy check** means that TiDB does not check for the unique constraint when the `INSERT` statement is executed, but checks for the unique constraint when the transaction is committed. In TiDB, the lazy check is performed for values written by ordinary `INSERT` statements.
```

Reviewer 2: I think it is more accurate to title "Lazy checking of constraints".

> Example:

Reviewer: ```suggestion
For example:
```

```sql
CREATE TABLE T (I INT KEY);
INSERT INTO T VALUES (1);
BEGIN;

> INSERT INTO T VALUES (1); -- MySQL returns an error; TiDB returns success

Reviewer: ```suggestion
INSERT INTO T VALUES (1); -- MySQL returns an error; TiDB returns success.
```

INSERT INTO T VALUES (2);

> COMMIT; -- MySQL commits successfully; TiDB returns an error, transaction rollback

Reviewer: ```suggestion
COMMIT; -- It is successfully committed in MySQL; TiDB returns an error and the transaction rolls back.
```

> SELECT * FROM T; -- MySQL returns 1 2; TiDB returns 1

Reviewer: ```suggestion
SELECT * FROM T; -- MySQL returns 1 2; TiDB returns 1.
```

Reviewer: For code comment style, please follow this guide: https://github.com/pingcap/community/blob/master/code-comment-style.md

```

> The significance of lazy checking is that: if you perform a unique constraint check on every `INSERT` statement in a transaction, it will cause high network overhead. A batch check at the time of submission will greatly improve performance.

Reviewer: ```suggestion
The lazy check is important because if you perform a unique constraint check on every `INSERT` statement in a transaction, it can cause high network overhead. A batch check when the transaction is committed can greatly improve performance.
```

> > **Note**: This optimization does not take effect for `INSERT IGNORE` and `INSERT ON DUPLICATE KEY UPDATE`, only for normal `INSERT` statements.

Reviewer: - Please leave a blank line before the Note.
- Keep wording consistent.

```suggestion
> **Note:** This optimization does not take effect for `INSERT IGNORE` and `INSERT ON DUPLICATE KEY UPDATE` statements, and only takes effect for ordinary `INSERT` statements.
```
